code/addons/interactions/template/stories/basics.stories.ts

```diff
@@ -0,0 +1,75 @@
+/* eslint-disable jest/no-standalone-expect */
+import globalThis from 'global';
+import { within, waitFor, fireEvent, userEvent } from '@storybook/testing-library';
+import { expect } from '@storybook/jest';
+
+export default {
+  component: globalThis.Components.Form,
+  argTypes: {
+    onSuccess: { type: 'function' },
+  },
+};
+
+export const Type = {
+  play: async ({ canvasElement }) => {
+    const canvas = within(canvasElement);
+    await userEvent.type(canvas.getByTestId('value'), 'test');
+  },
+};
+
+export const Step = {
+  play: async (context) => {
+    await context.step('Enter value', async () => Type.play(context));
+  },
+};
+
+export const Click = {
+  play: async ({ canvasElement }) => {
+    const canvas = within(canvasElement);
+    await fireEvent.click(canvas.getByTestId('succeed'));
+  },
+};
+
+export const Callback = {
+  play: async (context) => {
+    const { args, canvasElement, step } = context;
+    const canvas = within(canvasElement);
+    await step('Enter value', async () => Type.play(context));
+
+    await step('Click checkbox', async () => Click.play(context));
+
+    await step('Submit', async () => {
+      await fireEvent.click(canvas.getByRole('button'));
+    });
+
+    await expect(args.onSuccess).toHaveBeenCalled();
+  },
+};
+
+export const WaitFor = {
+  play: async (context) => {
+    const { args, canvasElement, step } = context;
+    const canvas = within(canvasElement);
+    await step('Enter value', async () => Type.play(context));
+
+    await step('Submit', async () => {
+      await fireEvent.click(canvas.getByRole('button'));
+    });
+
+    await expect(args.onSuccess).not.toHaveBeenCalled();
+
+    await waitFor(
+      async function () {
+        await canvas.getByTestId("Submitted 'test' when not allowed!");
+      },
+      { timeout: 2000 }
+    );
+  },
```

Comment:
```suggestion
  play: async ({ args, canvasElement, step }) => {
    const canvas = within(canvasElement);

    await step('Enter value', Type.play);

    await step('Submit', async () => {
      await fireEvent.click(canvas.getByRole('button'));
    });

    await expect(args.onSuccess).not.toHaveBeenCalled();

    await waitFor(
      () => canvas.getByTestId("Submitted 'test' when not allowed!"),
      { timeout: 2000 }
    );
  },
```

Reply:
I guess you're intentionally using `waitFor` here, but the idiomatic solution would actually be this:
```js
await canvas.findByTestId("Submitted 'test' when not allowed!", { timeout: 2000 });
```
This does exactly the same thing, except that you don't get a waitFor block in the interactions panel.

Weird test ID by the way.

Reply:
Yes, I am aware. I really wanted to get a `waitFor` in here though (!)

> Weird test ID by the way

🤔